docs/family_app.rst

```diff
@@ -0,0 +1,116 @@
+:doc:`/blacklist_app`.
+
+By organizing tokens into families, you gain finer control over user sessions and potential compromises.
+For example, when the settings ``BLACKLIST_AFTER_ROTATION`` and ``TOKEN_FAMILY_BLACKLIST_ON_REUSE`` are
+set to ``True``, if a refresh token is stolen and used by both the valid user and the attacker,
+the system will detect the reuse of the token and automatically blacklist the associated family. 
+This invalidates every token that shares the same ``family_id`` as the reused refresh token, effectively
+cutting off access without waiting for individual token expiration.
+
+-------
+
+Simple JWT includes an app that provides token family functionality.  To use
+this app, include it in your list of installed apps in ``settings.py``:
+
+.. code-block:: python
+
+  # Django project settings.py
+
+  ...
+
+  INSTALLED_APPS = (
+      ...
+      'rest_framework_simplejwt.token_family',
+      ...
+  )
+
+Also, make sure to run ``python manage.py migrate`` to run the app's
+migrations.
+
+If the token family app is detected in ``INSTALLED_APPS`` and the setting
+``TOKEN_FAMILY_ENABLED`` is set to ``True``, Simple JWT will add a new family
+to the family list and will also add two new claims, "family_id" and 
+"family_exp", to the refresh tokens. It will also check that the
+family indicated in the token's payload does not appear in a blacklist of
+families before it considers it as valid, and it will also check that the
+family expiration date ("family_exp") has not passed; if the family is expired,
+then the token will be considered as invalid.
+
+The Simple JWT family app implements its family and blacklisted family
+lists using two models: ``TokenFamily`` and ``BlacklistedTokenFamily``.  Model
+admins are defined for both of these models.  To add a family to the blacklist,
+find its corresponding ``TokenFamily`` record in the admin and use the
+admin again to create a ``BlacklistedTokenFamily`` record that points to the
+``TokenFamily`` record.
+
+Alternatively, you can blacklist a family by creating a ``FamilyMixin``
+subclass instance and calling the instance's ``blacklist_family`` method:
+
+.. code-block:: python
+
+  from rest_framework_simplejwt.tokens import RefreshToken
+
+  token = RefreshToken(base64_encoded_token_string)
+  token.blacklist_family()
+
+Keep in mind that the ``base64_encoded_token_string`` should already
+contain a family ID claim in its payload.
+
+This will create a unique family and blacklist records for the token's
+"family_id" claim or whichever claim is specified by the ``TOKEN_FAMILY_CLAIM`` setting.
+
+
+In a ``urls.py`` file, you can also include a route for ``TokenFamilyBlacklistView``:
+
+.. code-block:: python
+
+  from rest_framework_simplejwt.views import TokenFamilyBlacklistView
+
+  urlpatterns = [
+    ...
+    path('api/token/family/blacklist/', TokenFamilyBlacklistView.as_view(), name='token_family_blacklist'),
```

Comment:
Why would we want to use a different view for token family blacklist? Shouldn't this simply replace the need for the Blacklist url itself?